allchblk.c

```diff
@@ -984,6 +989,34 @@ GC_INNER void GC_freehblk(struct hblk *hbp)
           hhdr = prevhdr;
         }
       }
+    #else  /* !defined(__CHERI_PURE_CAPABILITY__) */
+    /* Coalesce with successor, if possible */
+      if(0 != nexthdr && HBLK_IS_FREE(nexthdr) && IS_MAPPED(nexthdr)
+         && (signed_word)(hhdr -> hb_sz + nexthdr -> hb_sz) > 0
+	 && (cheri_base_get(hbp) <= cheri_address_get(next))
+	 && (cheri_base_get(hbp) + cheri_length_get(hbp)) >= (cheri_address_get(next) + nexthdr->hb_sz)
```

Comment:
Is the length of `hbp` expected to be precise, or are imprecise bounds here OK?

Reply:
imprecise bounds are OK here. This is memory being reclaimed into memory pools for the allocator to dole out later. 

I believe the narrowing of bounds (as precise as possible) is much more accurate when done by the allocator during each memory request. 

During the sweeping stage, we are only trying to ensure that any capability can access all the memory that is being coalesced together. When the next call into this function happens from a contiguous free memory chunk, it will start checking from the derived pointer from the previous call. So it will  again  check for capability bounds being able to access all the memory being coalesced.

Reply:
I guess what I'm worried about is if the bounds are really big, might this check accidentally cover more than one block?

Reply:
I think a comment in the code explaining some of this would be very helpful!

Reply:
added small comment in [632aa0c](https://github.com/capablevms/bdwgc/pull/8/commits/632aa0c72a478ccf9fc201842578f7d35b822166).